hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/management/JpaTargetManagement.java

```diff
@@ -542,49 +539,45 @@ public TargetTypeAssignmentResult assignType(final Collection<String> controller
     @Transactional
     @Retryable(include = {
             ConcurrencyFailureException.class }, maxAttempts = Constants.TX_RT_MAX, backoff = @Backoff(delay = Constants.TX_RT_DELAY))
-    public TargetTypeAssignmentResult unAssignType(final Collection<String> controllerIds) {
+    public TargetTypeAssignmentResult unassignType(final Collection<String> controllerIds) {
         final List<JpaTarget> allTargets = findTargetsByInSpecification(controllerIds, null);
 
         if (allTargets.size() < controllerIds.size()) {
             throw new EntityNotFoundException(Target.class, controllerIds,
-                    allTargets.stream().map(Target::getControllerId).collect(Collectors.toList()));
+                    allTargets.stream().map(Target::getControllerId).toList());
         }
 
         // set new target type to null for all targets
         allTargets.forEach(target -> target.setTargetType(null));
 
-        return new TargetTypeAssignmentResult(0, Collections.emptyList(), Collections
-                .unmodifiableList(allTargets.stream().map(targetRepository::save).collect(Collectors.toList())), null);
+        return new TargetTypeAssignmentResult(0, Collections.emptyList(), targetRepository.saveAll(allTargets), null);
     }
 
     private List<JpaTarget> findTargetsByInSpecification(final Collection<String> controllerIds,
             final Specification<JpaTarget> specification) {
         return Lists.partition(new ArrayList<>(controllerIds), Constants.MAX_ENTRIES_IN_STATEMENT).stream()
```

Comment:
What about using a parallel stream here?

```suggestion
        return Lists.partition(new ArrayList<>(controllerIds), Constants.MAX_ENTRIES_IN_STATEMENT).parallelStream()
```

Reply:
I'm not sure. This could occupy more threads. And anyway is not part of this PR. Maybe could evaluate such changes later.